django/thunderstore/frontend/models.py

```diff
@@ -84,4 +85,51 @@ def get_for_community(cls, community: Optional[Community], placement: str):
         return cls.objects.filter(full_query).order_by("-ordering", "-pk")
 
 
+class LinkTargetChoices(TextChoices):
+    Blank = "_blank"
+    Parent = "_parent"
+    Self = "_self"
+    Top = "_top"
+
+
+class NavLinkMixin(TimestampMixin):
+    title = models.TextField()
+    href = models.TextField()
+    css_class = models.TextField(blank=True, null=True)
+    target = models.TextField(
```

Comment:
There's no practical difference between a `CharField` and a `TextField` in postgres, it just makes the max length field required on Django's side and changes the admin widget. So I generally prefer using `TextField` instead of pretending there's a difference.